### PR TITLE
[AMDGPU] Add documentation for scheduler intrinsics

### PR DESCRIPTION
**Pull request by bcahoon**
Adding sched_barrier, sched_group_barrier, and iglp_opt.

### CODE REVIEW

**Review by Rmalavally — APPROVED** [review]

**Review by arsenm — APPROVED** [review]

File: llvm/docs/AMDGPUUsage.rst [hunk 1]
```diff
@@ -1098,6 +1098,53 @@ The AMDGPU backend implements the following LLVM IR intrinsics.
...
+
+                                                   |  ``// 1 VMEM read``
+                                                   |  ``__builtin_amdgcn_sched_group_barrier(32, 1, 0)``
+                                                   |  ``// 1 VALU``
+                                                   |  ``__builtin_amdgcn_sched_group_barrier(2, 1, 0)``
+                                                   |  ``// 5 MFMA``
+                                                   |  ``__builtin_amdgcn_sched_group_barrier(8, 5, 0)``
+
+  llvm.amdgcn.iglp_opt                             An **experimental** intrinsic for instruction group level parallelism. The intrinsic
+                                                   implements predefined intruction scheduling orderings. The intrinsic applies to the
```
Comment (jrbyrnes) [inline llvm/docs/AMDGPUUsage.rst:1140]: It should apply to the basic block, independent of its location.

File: llvm/docs/AMDGPUUsage.rst [hunk 2]
```diff
@@ -1098,6 +1098,53 @@ The AMDGPU backend implements the following LLVM IR intrinsics.
                                                    with the fifth i32 operand. The i1 sixth operand is used to clamp
                                                    the output. The i1s preceding the vector operands decide the signedness.
 
+  llvm.amdgcn.sched_barrier                        Controls the types of instructions that may be allowed to cross the intrinsic
+                                                   during instruction scheduling. The parameter is a mask for the instruction types
+                                                   that can cross the intrinsic.
+
```
Comment (jrbyrnes) [inline llvm/docs/AMDGPUUsage.rst:1104]: Perhaps a note that the mask is additive (and not negatable)? E.g. 0x0006 Allows VALU or SALU. 0x03FC allows everything but VALU. But ~(0x0002) is an error. 

Similar for sched_group_barrier, 0x0006 creates schedgroup which can contain either VALU or SALU

File: llvm/docs/AMDGPUUsage.rst [hunk 3]
```diff
@@ -1098,6 +1098,53 @@ The AMDGPU backend implements the following LLVM IR intrinsics.
...
+  llvm.amdgcn.iglp_opt                             An **experimental** intrinsic for instruction group level parallelism. The intrinsic
+                                                   implements predefined intruction scheduling orderings. The intrinsic applies to the
+                                                   code that appears after the intrinsic. The intrinsic takes a value that specifies the
+                                                   strategy.  The compiler implements two strategies.
+
+                                                   0. Interleave DS and MFMA instructions for small GEMM kernels.
+                                                   1. Interleave DS and MFMA instructions for single wave small GEMM kernels.
+
+                                                   The iglp_opt strategy implementations are subject to change.
```
Comment (jrbyrnes) [inline llvm/docs/AMDGPUUsage.rst:1147]: iglp_opt is mutually exclusive with sched_barrier + sched_group_barrier.

**Review by jrbyrnes — COMMENTED** [review]
Descriptions LGTM with a few details.

**Review by jrbyrnes — APPROVED** [review]
LGTM

**Pull request merged**